pkg/agent/datamodel/types.go

```diff
@@ -1000,6 +1000,14 @@ func (o *OrchestratorProfile) IsAzureCNI() bool {
 	return false
 }
 
+// IsByoCNI returns true if Byo CNI network plugin is enabled
+func (o *OrchestratorProfile) IsByoCNI() bool {
+	if o.KubernetesConfig != nil {
+		return strings.EqualFold(o.KubernetesConfig.NetworkPlugin, NetworkPluginByo)
```

Comment:
tagentially, can this ever be empty? e.g. for default azure cni?

Reply:
Nope we're pretty strict about only allowing in either `--network-plugin kubenet` or `--network-plugin azure`. After this is available in AgentBaker, I will make the adjustments in RP to allow `--network-plugin byo`.

Reply:
And if the flag isn't given we'll always default to `kubenet` and set it up like that.